### ISSUE TITLE
Lazily load/unload relooper

### ISSUE THREAD

**Issue by EricRahm** [mozilla/j2me.js#1129]
We have at lest 4.3MB of asm.js related ArrayBuffers, in theory we should be able to reduce that by either not loading the relooper until it's needed (so after having run a few times), or unloading the relooper after a period of inactivity.

**Comment by andreasgal** [comment]
The relooper should not be needed most the time since we cache compiled code, right? We definitely must load it lazily.

**Comment by mbebenita** [comment]
Yes, it should be lazy. However, our cache only stabilizes after several runs. We can tweak the heurisitcs to compile code after a larger number of calls/iterations, at the moment, that number is 10. The larger the number, the less we compile and less often.

Other things to try to reduce memory:

Currently, we don't compile methods whose code size is larger than 2000 bytes. This is probably too large, we may want to experiment with a lower number, or other heuristics.

At the moment, the relooper has a 4MB heap. This is used for internal data structures, stack space and the resulting output buffer into which the relooper writes the generated code. We can tweak these numbers https://github.com/mozilla/j2me.js/blob/master/jit/relooper/Makefile#L5 and https://github.com/mozilla/j2me.js/blob/master/jit/relooper/Makefile#L8
